lib/compilers/tasking.ts

```diff
@@ -0,0 +1,124 @@
+import path from 'path';
+import {fileURLToPath} from 'url';
+
+import {CompilationResult, ExecutionOptions} from '../../types/compilation/compilation.interfaces';
+import {CompilerOutputOptions, ParseFiltersAndOutputOptions} from '../../types/features/filters.interfaces';
+import {BaseCompiler} from '../base-compiler';
+import {AsmParserTasking} from '../parsers/asm-parser-tasking';
+
+export class TaskingCompiler extends BaseCompiler {
+    override asm: AsmParserTasking;
+    filtersBinary: boolean;
+
+    static get key() {
+        return 'tasking';
+    }
+
+    constructor(info, env) {
+        super(info, env);
+        this.asm = new AsmParserTasking(this.compilerProps);
+        const index = this.compiler.exe.indexOf('cctc.exe');
+        this.compiler.exe = this.compiler.exe.slice(0, index) + 'hldumptc.exe';
+    }
+
+    override optionsForFilter(filters, outputFilename) {
+        //hldumptc -cc -FCdFHMNSY -is $(OUTPATH)/$< -o $(OUTPATH)/cppdemo.asm
+        if (filters.binary) this.filtersBinary = true;
+        else this.filtersBinary = false;
+
+        return ['-cc', '-FCdFHMNSY', '-o', this.filename(outputFilename)];
+    }
+
+    override async runCompiler(compiler, options, inputFilename, execOptions) {
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        execOptions.customCwd = path.dirname(inputFilename);
+
+        const index = compiler.indexOf('hldumptc.exe');
+        const compilertasking = compiler.slice(0, index) + 'cctc.exe';
+        const optionstasking = ['1', '2', '3', '4'];
```

Comment:
better use String.replace()
what is const optionstasking = ['1', '2', '3', '4']; for ?